app/screens/integration_selector/index.tsx

```diff
@@ -0,0 +1,609 @@
+                return;
+            }
+            case ViewConstants.DATA_SOURCE_CHANNELS: {
+                if (currentSelected[itemKey]) {
+                    delete multiselectSelectedItems[itemKey];
+                } else {
+                    multiselectSelectedItems[itemKey] = item as Channel;
+                }
+
+                setMultiselectSelected(multiselectSelectedItems);
+                return;
+            }
+            default: {
+                if (currentSelected[itemKey]) {
+                    delete multiselectSelectedItems[itemKey];
+                } else {
+                    multiselectSelectedItems[itemKey] = item as DialogOption;
+                }
+                setMultiselectSelected(multiselectSelectedItems);
+            }
+        }
+    }, [integrationData, multiselectSelected, isMultiselect, dataSource, handleSelect]);
+
+    const handleRemoveOption = (item: UserProfile | Channel | DialogOption) => {
+        const currentSelected: Dictionary<UserProfile> | Dictionary<DialogOption> | Dictionary<Channel> = multiselectSelected;
+        const itemKey = extractItemKey(dataSource, item);
+        const multiselectSelectedItems = {...currentSelected};
+        delete multiselectSelectedItems[itemKey];
+        setMultiselectSelected(multiselectSelectedItems);
+    };
+
+    const getChannels = useCallback(debounce(async () => {
+        if (next.current && !loading && !term) {
+            setLoading(true);
+            page.current += 1;
+
+            const {channels: channelData} = await fetchChannels(serverUrl, currentTeamId, page.current);
+
+            setLoading(false);
+
+            if (channelData && channelData.length > 0) {
+                setIntegrationData(channelData);
+            } else {
+                next.current = false;
+            }
+        }
+    }, 100), [integrationData, loading, term]);
+
+    const getProfiles = useCallback(debounce(async () => {
+        if (next.current && !loading && !term) {
+            setLoading(true);
+            page.current += 1;
+
+            const {users: profiles} = await fetchProfiles(serverUrl, page.current);
+
+            setLoading(false);
+
+            if (profiles && profiles.length > 0) {
+                setIntegrationData(profiles);
+            } else {
+                next.current = false;
+            }
+        }
+    }, 100), [integrationData, loading, term]);
+
+    const loadMore = useCallback(async () => {
+        if (dataSource === ViewConstants.DATA_SOURCE_USERS) {
+            await getProfiles();
+        } else if (dataSource === ViewConstants.DATA_SOURCE_CHANNELS) {
+            await getChannels();
+        }
+
+        // dynamic options are not paged so are not reloaded on scroll
+    }, [getProfiles, getChannels, dataSource]);
+
+    const searchDynamicOptions = useCallback(async (searchTerm = '') => {
+        if (options) {
+            setIntegrationData(options);
```

Comment:
This seems weird. Every time you call searchDynamicOptions, if the options are present, you overwrite the integration data. When does this route happen? When it is not dynamic but static data? Or should never happen?

The main issue here is that for every change in the term we are reassigning the state. Hopefully is not a huge deal because we are passing the same object always, but it probably still generates a re-render of this object (not sure 100%). Would be great if you could check and update this.

Also, I see that you show the options, then search for the dynamic options, and then overwrite these options on line 281. It seems weird.

Reply:
Fixed